### DISCUSSION TITLE
Delete unwanted artifacts

### DISCUSSION THREAD

**Discussion by chris6713** [ihhub/fheroes2#9334] [Ideas]
I ( and many others ) have been hoping for a way to clean out unwanted artifacts from my heroes inventory. The technique of transferring the items to a mule hero and then dismissing that hero has issues, not only is it time consuming but the AI likes to hire those heroes and then you just get those artifacts back when you end up fighting said hero. Not to mention eventually you start running out of mule heroes on bigger maps ( yes I speak from experience ).

I think a reasonable option is to just be able to control click an item in a heroes inventory and have that item disappear. I've already done this for the PC version of fheroes 2 on my computer by editing artifact.cpp, took only a few lines of code. For me this is a huge QOL improvement.

Ideally there would be a popup "Are you sure you want to delete this artifact" confirmation box, and certain "curse" items can be made immune to deletion. However it's a pretty easy fix with few changes required.

**Comment by oleg-derevenetz** [comment]
This is not a question of QoL, but a question of a core game mechanics. The whole point of the cursed artifacts is to force you either to spend a money to hire a "mule hero" or to use a special building on the map (Alchemist's Tower). Ability to just remove the unwanted artifacts by a few clicks is kind of ruins all that. So no, it won't happen. Sure, you can alter any game mechanics for yourself, but changes of this kind are unlikely to get into the `master` branch.

**Reply by chris6713** [reply]
I'm not talking about cursed artifacts. I specifically stated **curse" items can be made immune to deletion**.
This is about the many copies of unwanted artifacts like medals, minor scrolls, coins, rabbit feet, skull cap, evil eye etc... that will overload you and your mules on larger maps ( especially with water and lots of shipwrecked sailors ).

**Reply by oleg-derevenetz** [reply]
Once again, this is not the question of QoL. Artifact management is considered as a part of the core game mechanics, a player who is able to do it more efficiently gets an advantage.

**Reply by chris6713** [reply]
**Artifact management is considered as a part of the core game mechanics**

Say's who? And you seem to be thinking of only multiplayer here? I am thinking of single player.

You can make the same **core mechanic** argument about many of the QoL improvements Fheroes2 makes already.

I'd like to hear a solid argument of why clearing out unwanted artifacts by transferring them to mules is a part of core game mechanics. To me it seems more like a developer oversight ( rectified by later HOMM titles with the artifact merchant ).

**Reply by oleg-derevenetz** [reply]
> I'd like to hear a solid argument of why clearing out unwanted artifacts by transferring them to mules is a part of core game mechanics.

Because that's how it works in the original game. It's that simple. You can consider this as a "developer oversight", I will consider this as a part of the OG mechanics and I'll vote against changing this. You can make whatever changes you want in your own fork.

**Reply by chris6713** [reply]
**Because that's how it works in the original game**

Many other things have been changed from the original game for purposes of QoL, so to me that's not a convincing argument. But if that's your answer I guess I will consider forking it.

**Reply by oleg-derevenetz** [reply]
> Many other things have been changed from the original game for purposes of QoL.

There were a lot more of questionable changes from the previous team, but most of them were removed before 1.0 release. Some changes that affect the game mechanics are still left (e.g. spacebar mechanics or ability to buy a non-upgraded creatures in an upgraded building), but this doesn't mean that we will introduce a new changes in a game mechanics lightly. 

> so to me that's not a convincing argument.

This is not a technical question where some approach will work and other approach will not. This is an opinion-based question.

**Reply by Branikolog** [reply]
Hello, @chris6713 !
Let me join the discussion here. 😃 
I do agree with fheroes2 dev team in this question. The original mechanic implies, that you're enforced to have some free space in your inventory to get new artifacts, otherwise you can lose those you've captured from defeated enemy. So there's always a compromise between taking with you the most useful artifacts and leaving some free space for the potential new ones.
Unfortunately, the original mechanic has no other solution to get rid of possessed artifacts, rather than transferring them to your other hero. I understand an agree with you, that this could be annoying. But completely removing this core game mechanics, allowing player instantly removing everything from your inventory is not the best solution of the problem. Also, cursed artifacts somehow being immune to the deletion feature looks not intuitive (You can delete every item from you pocket, except of the cursed one. Why?) We just loose an essential part of the gameplay with such approach.
I think, we can use similar approach that was implemented by the original dev team: a special map object that allows you to interact with some artifacts from your inventory.
In my opinion adding a new unique map structure that allows you to get rid of any positive artifact is the most safe solution of overfilled inventory on the late game stage. Any player, can visit a special place and get rid of unwanted artifacts according to the custom rule of the map structure. It could be some exp or 100-200 gold per each item. At the same time player still has to consider the limited space of the inventory, but he can plan his strategy according to the location of this new map structure. It could also be a special town structure or additional marketplace feature. But allowing any hero instantly deleting all legendary items from his inventory during his quest is a big "no" from my side.

**Reply by chris6713** [reply]
> Hello, @chris6713 ! Let me join the discussion here. 😃 I do agree with fheroes2 dev team in this question. The original mechanic implies, that you're enforced to have some free space in your inventory to get new artifacts, otherwise you can lose those you've captured from defeated enemy. So there's always a compromise between taking with you the most useful artifacts and leaving some free space for the potential new ones. Unfortunately, the original mechanic has no other solution to get rid of possessed artifacts, rather than transferring them to your other hero. I understand an agree with you, that this could be annoying. But completely removing this core game mechanics, allowing player instantly removing everything from your inventory is not the best solution of the problem. Also, cursed artifacts somehow being immune to the deletion feature looks not intuitive (You can delete every item from you pocket, except of the cursed one. Why?) We just loose an essential part of the gameplay with such approach. I think, we can use similar approach that was implemented by the original dev team: a special map object that allows you to interact with some artifacts from your inventory. In my opinion adding a new unique map structure that allows you to get rid of any positive artifact is the most safe solution of overfilled inventory on the late game stage. Any player, can visit a special place and get rid of unwanted artifacts according to the custom rule of the map structure. It could be some exp or 100-200 gold per each item. At the same time player still has to consider the limited space of the inventory, but he can plan his strategy according to the location of this new map structure. It could also be a special town structure or additional marketplace feature. But allowing any hero instantly deleting all legendary items from his inventory during his quest is a big "no" from my side.

The map object was fine with me but I was thinking that would be rejected even harder because it's creating something that was never there before originally. But I'm OK with the idea. The fact that you can literally NEVER get rid of the artifacts, just shuffle them around was the part that struck me as an oversight.

**Reply by oleg-derevenetz** [reply]
In general, I'm not against the special building, because:

* This does not violate the general canon of the game, according to which you can't just get rid of artifacts. You have to spend something - money, movement points, time, and so on.
* A building with a similar purpose already exists in the original game.
* This will only apply to FH2M maps. The innovation will not affect the original maps and campaigns.

But this requires the work of an artist, as a new object is needed, and most likely with variants for different terrains.

**Reply by chris6713** [reply]
> In general, I'm not against the special building, because:
> 
> * This does not violate the general canon of the game, according to which you can't just get rid of artifacts. You have to spend something - money, movement points, time, and so on.
> * A building with a similar purpose already exists in the original game.
> * This will only apply to FH2M maps. The innovation will not affect the original maps and campaigns.
> 
> But this requires the work of an artist, as a new object is needed, and most likely with variants for different terrains.

What would be the process of submitting art for this?

**Reply by Mr-Bajs** [reply]
An on map building would be great. An auction house, where you could buy and sell artifacts once a week.

**Reply by zenseii** [reply]
@chris6713. Here is our guide which will help get you started: https://github.com/ihhub/fheroes2/blob/master/docs/GRAPHICAL_ASSETS.md. You can contact either @ihhub, @Branikolog or me with a direct message or in the channel dedicated to art on our Discord server.
